### DISCUSSION TITLE
Does nextjs build typescript too with js files?

### DISCUSSION THREAD

**Discussion by nokdoot** [vercel/next.js#16963] [Help] [answered]
I'm about to use custom server way and project has ts for server(db init too), js for browser(react) and run project with 'node server.js'
But I'm consider how to build and how to run typescript imported by server.'js' because next build makes on .next directory

**Comment by jamesmosier** [comment, marked as answer]
If you want to use a custom server with typescript then the [custom-server-typescript](https://github.com/vercel/next.js/tree/canary/examples/custom-server-typescript) example is a great place to look. You can then use regular JS for your browser (react) components if you'd like.
